crates/ide-diagnostics/src/handlers/unresolved_field.rs

```diff
@@ -46,15 +50,88 @@ pub(crate) fn unresolved_field(
 }
 
 fn fixes(ctx: &DiagnosticsContext<'_>, d: &hir::UnresolvedField) -> Option<Vec<Assist>> {
-    if d.method_with_same_name_exists {
-        method_fix(ctx, &d.expr)
-    } else {
-        // FIXME: add quickfix
-
-        None
+    let mut fixes = if d.method_with_same_name_exists { method_fix(ctx, &d.expr) } else { None };
+    if let Some(fix) = add_field_fix(ctx, d) {
+        fixes.get_or_insert_with(Vec::new).push(fix);
     }
+    fixes
 }
 
+fn add_field_fix(ctx: &DiagnosticsContext<'_>, d: &hir::UnresolvedField) -> Option<Assist> {
+    // Get the FileRange of the invalid field access
+    let root = ctx.sema.db.parse_or_expand(d.expr.file_id);
+    let expr = d.expr.value.to_node(&root);
+
+    let error_range = ctx.sema.original_range_opt(expr.syntax())?;
+    // Convert the receiver to an ADT
+    let adt = d.receiver.as_adt()?;
+    let Adt::Struct(adt) = adt else {
+        return None;
+    };
+
+    let target_module = adt.module(ctx.sema.db);
+
+    let suggested_type =
+        if let Some(new_field_type) = ctx.sema.type_of_expr(&expr).map(|v| v.adjusted()) {
+            let display =
+                new_field_type.display_source_code(ctx.sema.db, target_module.into(), true).ok();
+            make::ty(display.as_deref().unwrap_or("()"))
+        } else {
+            make::ty("()")
+        };
+
+    if !is_editable_crate(target_module.krate(), ctx.sema.db) {
+        return None;
+    }
+    let adt_source = adt.source(ctx.sema.db)?;
+    let adt_syntax = adt_source.syntax();
+    let range = adt_syntax.original_file_range(ctx.sema.db);
+
+    // Get range of final field in the struct
+    let (offset, needs_comma, indent) = match adt.fields(ctx.sema.db).last() {
+        Some(field) => {
+            let last_field = field.source(ctx.sema.db)?.value;
```

Comment:
```suggestion
            let last_field = adt_syntax.field_list()?.fields().last();
```
somethign like that, no need to go via `source` since we have the main source already